gb-cpu/src/microcode/logic.rs

```diff
@@ -19,3 +19,30 @@ pub fn xor(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
     ctl.push(value);
     OK_PLAY_NEXT_ACTION
 }
+
+pub fn cpl(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
+    let value = ctl.pop();
+
+    ctl.push(!value);
+    state.regs.set_half_carry(true);
+    state.regs.set_subtraction(true);
+    OK_PLAY_NEXT_ACTION
+}
+
+pub fn and(ctl: &mut MicrocodeController, state: &mut State) -> MicrocodeFlow {
+    let value = ctl.pop();
+    let value = ctl.pop() & value;
+    state.regs.set_raw(0);
+    state.regs.set_zero(value == 0);
```

Comment:
after reading the doc, `half carry` need to be set

Reply:
Actually it seems that most of the flags weren't set correctly.